client/src/components/util/truncate.js

```diff
@@ -0,0 +1,78 @@
+import React, { cloneElement } from "react";
+import { Tooltip, Position } from "@blueprintjs/core";
+
+import { tooltipHoverOpenDelayQuick } from "../../globals";
+
+export default (props) => {
+  const { children } = props;
+  // Truncate only support a single child with a text child
+
+  if (
```

Comment:
nit: for readability, we can early throw error if the expected condition is not met, so we can avoid nesting our main code block in a giant if condition! 

Like so:

```ts
export default () => {
  const { children } = props;

  // Take care of exception up front
  if (React.Children.count(children) !== 1 && React.Children.count(children.props?.children) !== 1) {
    throw Error("Only pass a single child with text to Truncate");
  }

  // main functionality starts here with no nesting!
  const originalString = children.props.children;
  
  ...
}
```